frontend/components/ShareButton/index.jsx

```diff
@@ -0,0 +1,69 @@
+import React, { Component } from 'react';
+import ShareIcon from '@shopgate/pwa-ui-ios/icons/ShareIcon';
+import MaterialShareIcon from '@shopgate/pwa-ui-material/icons/MaterialShareIcon';
+import Ripple from '@shopgate/pwa-ui-shared/Ripple';
+import PropTypes from 'prop-types';
+import styles from './style';
+import getConfig from '../../helpers/getConfig';
+import connect from '../../connector';
+
+const { androidShareSVG } = getConfig();
+const { iOSShareSVG } = getConfig();
+/**
+ * ShareButton component
+ */
+class ShareButton extends Component {
+  static propTypes = {
+    iOSTheme: PropTypes.func.isRequired,
+    shareItem: PropTypes.func.isRequired,
+    shareParams: PropTypes.shape(),
+  };
+
+  static defaultProps = {
+    shareParams: null,
+  };
+  /**
+   * Handles the share button click
+   * Show's share screen for app
+   * @param {Object} event The click event object
+   */
+  handleClick = () => {
+    this.props.shareItem();
+  }
+  /**
+   * Renders the share icon depending on theme
+   * @returns {JSX}
+   */
+  renderIcon() {
+    if (this.props.iOSTheme()) {
+      return <ShareIcon svg={iOSShareSVG} />;
+    }
+    return <MaterialShareIcon svg={androidShareSVG} />;
+  }
+  /**
+   * Renders the components
+   * @returns {JSX}
+   */
+  render() {
+    if (!this.props.shareParams) {
+      return null;
+    }
+    const buttons = this.props.iOSTheme() ? styles.iOSButtons : styles.androidButtons;
+    const className = this.props.iOSTheme() ? styles.buttonFlat : styles.button;
+    return (
+      <div className={`${buttons}`}>
+        <button
+          className={`${className}`}
+          onClick={this.handleClick}
+          data-test-id="shareIcon"
+        >
+          <Ripple className={`${styles.ripple}`}>
+            {this.renderIcon()}
+          </Ripple>
```

Comment:
Also on GMD the button covers partially the image slider pagination.

![screenshot 2018-10-22 at 13 06 19](https://user-images.githubusercontent.com/3393569/47290102-2b51f100-d5fe-11e8-8d8d-68a09fb8aaa2.png)

 I don't have a good idea how to solve this problem, maybe put it to the right? Maybe PO should decide for this.
![screenshot 2018-10-22 at 13 10 09](https://user-images.githubusercontent.com/3393569/47290117-3e64c100-d5fe-11e8-934d-8aed67218599.png)

Reply:
Moved button to left side of screen. Not sure where best location would be.

Reply:
`right: 300` won't do the job. Try replacing it with `left: 0`.
![screenshot 2018-10-24 at 10 00 07](https://user-images.githubusercontent.com/3393569/47415418-abe62e00-d773-11e8-97a4-40ce109d86e0.png)

Reply:
iOS stroke looks fine, but I think the icon is somehow behind the header:
![screenshot 2018-10-24 at 10 09 34](https://user-images.githubusercontent.com/3393569/47415975-f61bdf00-d774-11e8-8a74-2f839b1b14c9.png)

Reply:
Set the zIndex to the same as the cta buttons in the iOS theme. Increasing the index didn't do anything for the appearance. Any tips? :/

Reply:
Modified Viewbox for iOS ShareIcon to show full svg

Reply:
That's not really fixing it. I compared the fav icon and share icon and made them have similar style:
https://github.com/shopgate/ext-pdp-native-share/pull/1/commits/b3fddc75f3da0716f7a761413dac0ee682aacfde

Reply:
For the problem with cutting off the bottom of the icon -> svg is broken, I'll fix that

Reply:
https://github.com/shopgate/pwa/pull/332/commits/4eaceb4ca8d9653df736ef51bdc50c4907e1faa9

Reply:
Please discuss button positions with me in the future if there are issues with the placement.